display-scene-from-mobile-scene-package/src/main/java/com/esri/arcgismaps/sample/displayscenefrommobilescenepackage/components/SceneViewModel.kt

```diff
@@ -0,0 +1,82 @@
+/* Copyright 2023 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.displayscenefrommobilescenepackage.components
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import com.arcgismaps.mapping.ArcGISScene
+import com.arcgismaps.mapping.MobileScenePackage
+import com.esri.arcgismaps.sample.displayscenefrommobilescenepackage.R
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import java.io.File
+
+class SceneViewModel(
+    private val application: Application,
+    private val sampleCoroutineScope: CoroutineScope
+) : AndroidViewModel(application) {
+
+    // create a mutable state holder for the first scene
+    private val _firstScene = MutableStateFlow<ArcGISScene?>(null)
+
+    // expose a read-only state flow for observing changes to the first scene
+    val firstScene: StateFlow<ArcGISScene?> = _firstScene.asStateFlow()
+
+    // create a mutable state holder for the snackbar message
+    private val _snackbarMessage = MutableStateFlow<String?>(null)
+
+    // expose a read-only state flow for observing changes to the snackbar message
+    val snackbarMessage: StateFlow<String?> = _snackbarMessage.asStateFlow()
```

Comment:
For v200 we have moved away from using snackbars since compose snackbar would need explicit theming. Instead we have a dialog to display a message in the `samples-lib` library which could be used to display an error. Check out Analyze Hotspots sample for reference: 

```suggestion
    // create a ViewModel to handle dialog interactions
    val messageDialogVM: MessageDialogViewModel = MessageDialogViewModel()
```